BDArmory/Guidances/MissileGuidance.cs

```diff
@@ -608,6 +792,385 @@ public static Vector3 GetTerminalManeuveringTarget(Vector3 targetPosition, Vesse
         public static FloatCurve DefaultLiftCurve = null;
         public static FloatCurve DefaultDragCurve = null;
 
+        // The below curves and constants are derived from the lift and drag curves and will need to be re-calculated
+        // if these are changed
+      
+        const float TRatioInflec1 = 1.181181181181181f; // Thrust to Lift Ratio (at AoA of 30) where the maximum occurs
+        // after the 65 degree mark
+        const float TRatioInflec2 = 2.242242242242242f; // Thrust to Lift Ratio (at AoA of 30) where a local maximum no
+        // longer exists, above this every section must be searched
+
+        public static FloatCurve AoACurve = null; // Floatcurve containing AoA of (local) max acceleration
+        // for a given thrust to lift (at the max CL of 1.5 at 30 degrees of AoA) ratio. Limited to a max
+        // of TRatioInflec2 where a local maximum no longer exists
+
+        public static FloatCurve AoAEqCurve = null; // Floatcurve containing AoA after which the acceleration goes above
+        // that of the local maximums'. Only exists between TRatioInflec1 and TRatioInflec2.
+
+        public static FloatCurve gMaxCurve = null; // Floatcurve containing max acceleration times the mass (total force)
+        // normalized by q*S*GLOBAL_LIFT_MULTIPLIER for TRatio between 0 and TRatioInflec2. Note that after TRatioInflec1
+        // this becomes a local maxima not a global maxima. This is used to narrow down what part of the curve we should
+        // solve on.
+
+        // Linearized CL v.s. AoA curve to enable fast solving. Algorithm performs bisection using the fast calculations of the bounds
+        // and then performs a linear solve 
+        public static float[] linAoA = null;
+        public static float[] linCL = null;
+        public static float[] linSin = null;
+        public static float[] linSlope = null;
+        public static float[] linIntc = null;
+
+        public static float getGLimit(MissileLauncher ml, float thrust, float gLim, float margin)//, out bool gLimited)
+        {
+            bool gLimited = false;
+
+            // Force required to reach g-limit
+            gLim *= (float)(ml.vessel.totalMass * PhysicsGlobals.GravitationalAcceleration);
+
+            float maxAoA = ml.maxAoA;
+
+            float currAoA = maxAoA;
+
+            int interval = 0;
+
+            // Factor by which to multiply the lift coefficient to get lift, it's the dynamic pressure times the lift area times
+            // the global lift multiplier
+            float qSk = (float) (0.5f * ml.vessel.atmDensity * ml.vessel.srfSpeed * ml.vessel.srfSpeed) * ml.liftArea * BDArmorySettings.GLOBAL_LIFT_MULTIPLIER;
```

Comment:
For vacuum stuff it'll be returning based on the optimal, non-aerodynamic constants. If anything has `ml.liftArea = 0` then no guidance will work with it in the first place since it means there is no lift force.